pyro/distributions/gaussian_scale_mixture.py

```diff
@@ -0,0 +1,145 @@
+from __future__ import absolute_import, division, print_function
+import math
+
+import torch
+from torch.autograd import Function
+from torch.autograd.function import once_differentiable
+from torch.distributions import constraints, Categorical
+
+from pyro.distributions.torch_distribution import TorchDistribution
+from pyro.distributions.util import sum_leftmost
+
+
+class GaussianScaleMixture(TorchDistribution):
+    """
+    Mixture of Normal distributions with zero mean and diagonal covariance matrices.
+
+    That is, this distribution is a mixture with K components, where each
+    component distribution is a D-dimensional Normal distribution with zero
+    mean and a D-dimensional diagonal covariance matrix. The K different
+    covariance matrices are controlled by the parameters `scale` and `lambdas`.
+    That is, the covariance matrix of the k'th component is given by
+
+    Sigma_ii = (lambda_k * scale_i) ** 2
+
+    where `lambda`_k is a positive scale factor and `scale`_i are positive
+    scale parameters shared between all K components. The mixture weights are
+    controlled by a K-dimensional vector of softmax logits, `logits`. This
+    distribution implements pathwise derivatives for samples from the distribution.
+    This distribution does not currently support batched parameters.
+
+    See reference [1] for details on the implementations of the pathwise
+    derivative. Please consider citing this reference if you use the pathwise
+    derivative in your research.
+
+    [1] Pathwise Derivatives for Multivariate Distributions, Martin Jankowiak &
+    Theofanis Karaletsos. arXiv:1806.01856
+
+    Note that this distribution supports both even and odd dimensions, but the
+    former should be more a bit higher precision, since it doesn't use any erfs in
+    the backward call.
+
+    :param torch.tensor scale: D-dimensional vector of scale
+    :param torch.tensor logits: K-dimensional vector of logits
+    :param torch.tensor lambdas: K-dimensional matrix of sigma multipliers
```

Comment:
We generally try to prefer semantic names over greek. Consider renaming to something like
```py
coord_scale
component_scale
component_logits
```

Reply:
done